pkg/controller/migmigration/dvm.go

```diff
@@ -0,0 +1,166 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:       labels,
+			GenerateName: t.Owner.GetName() + "-",
+			Namespace:    migapi.VeleroNamespace,
+		},
+		Spec: migapi.DirectVolumeMigrationSpec{
+			SrcMigClusterRef:            t.PlanResources.SrcMigCluster.GetObjectReference(),
+			DestMigClusterRef:           t.PlanResources.DestMigCluster.GetObjectReference(),
+			PersistentVolumeClaims:      *pvcList,
+			CreateDestinationNamespaces: true,
+		},
+	}
+	t.setDirectVolumeMigrationOwnerReferences(dvm)
+	return dvm
+}
+
+func (t *Task) getDirectVolumeMigration() (*migapi.DirectVolumeMigration, error) {
+	// Get correlation labels
+	labels := t.Owner.GetCorrelationLabels()
+	labels[DirectVolumeMigrationLabel] = t.UID()
+	// Get DVM with label
+	client, err := t.getDestinationClient()
+	if err != nil {
+		return nil, err
+	}
+	list := migapi.DirectVolumeMigrationList{}
+	err = client.List(
+		context.TODO(),
+		k8sclient.MatchingLabels(labels),
+		&list)
+	if err != nil {
+		return nil, err
+	}
+	if len(list.Items) > 0 {
+		return &list.Items[0], nil
+	}
+
+	return nil, nil
+}
+
+// Check if the DVM has completed.
+// Returns if it has completed, why it failed, and it's progress results
+func (t *Task) hasDirectVolumeMigrationCompleted(dvm *migapi.DirectVolumeMigration) (completed bool, failureReasons, progress []string) {
+	totalVolumes := len(dvm.Spec.PersistentVolumeClaims)
+	successfulPods := 0
+	failedPods := 0
+	runningPods := 0
+	if dvm.Status.SuccessfulPods != nil {
+		successfulPods = len(dvm.Status.SuccessfulPods)
+	}
+	if dvm.Status.FailedPods != nil {
+		failedPods = len(dvm.Status.FailedPods)
+	}
+	if dvm.Status.RunningPods != nil {
+		runningPods = len(dvm.Status.RunningPods)
+	}
+
+	volumeProgress := fmt.Sprintf("%v total volumes; %v successful; %v running; %v failed", totalVolumes, successfulPods, runningPods, failedPods)
+	switch dvm.Status.Phase {
+	case dvmc.Started:
+		// TODO: Update this to check on the associated dvmp resources and build up a progress indicator back to
+		progress = append(progress, fmt.Sprintf("direct volume migration started at %v. ", dvm.Status.StartTimestamp))
+	case dvmc.Completed:
+		progress = append(progress, fmt.Sprintf("%v/%v volume migrations were successful", successfulPods, totalVolumes))
+		completed = true
+	case dvmc.MigrationFailed:
+		failureReasons = append(failureReasons, fmt.Sprintf("direct volume migration failed. %s", volumeProgress))
+		completed = true
+	default:
+		progress = append(progress, volumeProgress)
+	}
+	return completed, failureReasons, progress
+}
+
+func (t *Task) getDirectVolumeClaimList() *[]migapi.PVCToMigrate {
+	pvcList := []migapi.PVCToMigrate{}
+	for _, pv := range t.PlanResources.MigPlan.Spec.PersistentVolumes.List {
+		if pv.Selection.Action != migapi.PvCopyAction && pv.Selection.CopyMethod != migapi.PvFilesystemCopyMethod {
+			continue
+		}
+		accessModes := pv.PVC.AccessModes
+		/* TODO: When access mode selection is available override this
```

Comment:
Do we enable this now? Or wait until UI supports? Talking with @sseago it sounded like we want this supported now since it works in the API. If so I can follow up with a PR.

Reply:
@dymurray I think we'd better enable this now (or in a follow-up PR) since otherwise this would be a regression for something supported when using restic.